### PR TITLE
OCPBUGS-36696: Add details about how to remove OADP Operator after IBU

### PR DESCRIPTION
**Pull request by amolnar-gh**
Version(s): 4.16, 4.17

Issue: https://issues.redhat.com/browse/OCPBUGS-36696

Link to docs preview:

- https://79042--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/image_based_upgrade/cnf-image-based-upgrade-base.html
- https://79042--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/image_based_upgrade/preparing_for_image_based_upgrade/cnf-image-based-upgrade-install-operators.html
- https://79042--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/image_based_upgrade/ztp-image-based-upgrade.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ztp-image-based-upgrade-upgrade.adoc [hunk 1]
```diff
@@ -136,16 +136,25 @@ cgu-ibu-upgrade         31h   Completed   All clusters are compliant with all th
...
+- fileName: OadpOperatorStatus.yaml
+  policyName: "subscriptions-policy"
+  complianceType: mustnothave
+- fileName: OadpSecret.yaml
+  policyName: "subscriptions-policy"
+  complianceType: mustnothave
+- fileName: OadpBackupStorageLocationStatus.yaml
+  policyName: "subscriptions-policy"
+  complianceType: mustnothave
+- fileName: DataProtectionApplication.yaml
```
Comment (Missxiaoguo) [inline modules/ztp-image-based-upgrade-upgrade.adoc:157]: These are in site PGT

File: modules/ztp-image-based-upgrade-installing-oadp.adoc [hunk 2]
```diff
@@ -127,20 +122,20 @@ spec:
...
-    - fileName: custom-crs/OadpOperatorStatus.yaml
+    - fileName: OadpOperatorStatus.yaml
       policyName: "subscriptions-policy"
 [...]
 ----
 
 . Create the `DataProtectionApplication` CR and the S3 secret:
 
-.. Create the following CRs in your `source-crs/custom-crs` directory:
+.. Create the following CRs in your `source-crs` directory:
```
Comment (Missxiaoguo) [inline modules/ztp-image-based-upgrade-installing-oadp.adoc:138]: These CRs are available in the ztp container image as well

File: modules/ztp-image-based-upgrade-upgrade.adoc [hunk 3]
```diff
@@ -128,4 +128,54 @@ cgu-ibu-finalize        30h   Completed   All clusters are compliant with all th
...
+  policyName: "subscriptions-policy"
+  complianceType: mustnothave
+----
+
+.. Change the `complianceType` to `mustnothave` for the OADP Operator namespace, Operator group, and subscription in the site `PolicyGenTemplate` file.
++
+[source,yaml]
+----
+- fileName: OadpSecret.yaml
+  policyName: "subscriptions-policy"
```
Comment (Missxiaoguo) [inline modules/ztp-image-based-upgrade-upgrade.adoc:158]: They should be previously included in the `config-policy`
```suggestion
  policyName: "config-policy"
```

File: edge_computing/image_based_upgrade/ztp-image-based-upgrade.adoc [hunk 4]
```diff
@@ -37,6 +37,8 @@ include::modules/ztp-image-based-upgrade-upgrade.adoc[leveloffset=+1]
 
 * xref:../../edge_computing/image_based_upgrade/ztp-image-based-upgrade.adoc#ztp-image-based-upgrade-rollback_ztp-gitops[Moving to the Rollback stage of the image-based upgrade with {lcao} and {ztp}]
 
+* xref:../../edge_computing/cnf-talm-for-cluster-upgrades.adoc#talo-policies-concept_cnf-topology-aware-lifecycle-manager[Update policies on managed clusters]
```
Comment (dfitzmau) [inline edge_computing/image_based_upgrade/ztp-image-based-upgrade.adoc:40]: Outside the updates for this PR but somewhat related, but maybe remove "section" so it aligns with the Deleting Operators...sentence style: 

Screenshot from 2024-07-31 11-38-08

File: modules/ztp-image-based-upgrade-installing-lca.adoc [hunk 5]
```diff
@@ -7,13 +7,9 @@
...
 Install the {lcao} with {ztp-first} to do an image-based upgrade.
 
-.Prerequisites
-
-* Create a directory called `custom-crs` in the `source-crs` directory. The `source-crs` directory must be in the same location as the `kustomization.yaml` file.
-
 .Procedure
 
-. Create the following CRs in the `openshift-lifecycle-agent` namespace and push them to the `source-crs/custom-crs` directory:
+. Extract the following CRs from the `ztp-site-generate` container image and push them to `source-cr` directory:
```
Comment (dfitzmau) [inline modules/ztp-image-based-upgrade-installing-lca.adoc:12]:
```suggestion
. Extract the following CRs from the `ztp-site-generate` container image and push them to the `source-cr` directory:
```

File: modules/ztp-image-based-upgrade-upgrade.adoc [hunk 6]
```diff
@@ -128,4 +128,60 @@ cgu-ibu-finalize        30h   Completed   All clusters are compliant with all th
 cgu-ibu-prep            31h   Completed   All clusters are compliant with all the managed policies
 cgu-ibu-upgrade         31h   Completed   All clusters are compliant with all the managed policies
 ----
---
\ No newline at end of file
+--
+
+. You can remove the OADP Operator and its configuration files after a successful upgrade.
```
Comment (dfitzmau) [inline modules/ztp-image-based-upgrade-upgrade.adoc:133]: Global change. All OADP reference should be converted to the existing attributes:

Screenshot from 2024-07-31 11-43-45

Comment (dfitzmau) [reply modules/ztp-image-based-upgrade-upgrade.adoc:133]: From common-attributes.adoc

File: modules/ztp-image-based-upgrade-upgrade.adoc [hunk 7]
```diff
@@ -128,4 +128,60 @@ cgu-ibu-finalize        30h   Completed   All clusters are compliant with all th
...
+
+.. Apply the change to your target clusters by using the {cgu-operator-full}. For more information about rolling out configuration changes, see the "Update policies on managed clusters" section.
+
+.. Monitor the process. When the status of the `common-subscriptions-policy` and the `example-cnf-config-policy` policies for a target cluster are `Compliant`, the OADP Operator has been removed from the cluster. Get the status of the policies by running the following commands:
++
+[source,terminal]
+----
+$ oc get policy -n ztp-common common-subscriptions-policy
+----
+
```
Comment (dfitzmau) [inline modules/ztp-image-based-upgrade-upgrade.adoc:178]: Remove "+" to fix indentation issue:

Screenshot from 2024-07-31 11-45-19

**Review by dfitzmau — APPROVED** [review]
Hi @amolnar-rh . Nice work on this PR. Please squash the commits as I see 4 commits on this PR.

/remove-label peer-review-in-progress

/label peer-review-done

File: modules/ztp-image-based-upgrade-installing-oadp.adoc [hunk 8]
```diff
@@ -3,17 +3,13 @@
...
-
-.Prerequisites
-
-* Create a directory called `custom-crs` in the `source-crs` directory. The `source-crs` directory must be in the same location as the `kustomization.yaml` file.
+Install and configure the {oadp-short} Operator with {ztp} before starting the upgrade.
 
 .Procedure
 
-. Create the following CRs in the `openshift-adp` namespace for both the seed and target clusters, then push the CRs to the `source-crs/custom-crs` directory:
+. Extract the following CRs from the `ztp-site-generate` container image and push them to `source-cr` directory:
```
Comment (bhardesty) [inline modules/ztp-image-based-upgrade-installing-oadp.adoc:12]:
```suggestion
. Extract the following CRs from the `ztp-site-generate` container image and push them to the `source-cr` directory:
```

File: modules/ztp-image-based-upgrade-installing-oadp.adoc [hunk 9]
```diff
@@ -127,20 +122,20 @@ spec:
...
-    - fileName: custom-crs/OadpOperatorStatus.yaml
+    - fileName: OadpOperatorStatus.yaml
       policyName: "subscriptions-policy"
 [...]
 ----
 
 . Create the `DataProtectionApplication` CR and the S3 secret only for the target cluster:
 
-.. Create the following CRs in your `source-crs/custom-crs` directory:
+.. Extract the following CRs from the `ztp-site-generate` container image and push them to `source-cr` directory:
```
Comment (bhardesty) [inline modules/ztp-image-based-upgrade-installing-oadp.adoc:138]:
```suggestion
.. Extract the following CRs from the `ztp-site-generate` container image and push them to the `source-cr` directory:
```

File: modules/ztp-image-based-upgrade-upgrade.adoc [hunk 10]
```diff
@@ -128,4 +128,61 @@ cgu-ibu-finalize        30h   Completed   All clusters are compliant with all th
...
 ----
---
\ No newline at end of file
+--
+
+. You can remove the {oadp-short} Operator and its configuration files after a successful upgrade.
+
+.. Change the `complianceType` to `mustnothave` for the {oadp-short} Operator namespace, Operator group, and subscription in the `common-ranGen.yaml` file.
++
+[source,yaml]
```
Comment (bhardesty) [inline modules/ztp-image-based-upgrade-upgrade.adoc:137]: Is `common-ranGen.yaml` an OpenShift API object? If so, this and the following yaml examples should include the required fields (https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#required-fields).

File: modules/ztp-image-based-upgrade-upgrade.adoc [hunk 11]
```diff
@@ -128,4 +128,61 @@ cgu-ibu-finalize        30h   Completed   All clusters are compliant with all th
...
+  policyName: "config-policy"
+  complianceType: mustnothave
+- fileName: DataProtectionApplication.yaml
+  policyName: "config-policy"
+  complianceType: mustnothave
+----
+
+.. Merge the changes with your custom site repository and wait for the ArgoCD application to synchronize the change to the hub cluster. The status of the `common-subscriptions-policy` and the `example-cnf-config-policy` policies change to `Non-Compliant`.
+
+.. Apply the change to your target clusters by using the {cgu-operator-full}. For more information about rolling out configuration changes, see the "Update policies on managed clusters".
```
Comment (bhardesty) [inline modules/ztp-image-based-upgrade-upgrade.adoc:170]:
```suggestion
.. Apply the change to your target clusters by using the {cgu-operator-full}. For more information about rolling out configuration changes, see "Update policies on managed clusters".
```

**Review by bhardesty — COMMENTED** [review]
This looks great - I just had a couple very minor comments.

File: modules/ztp-image-based-upgrade-upgrade.adoc (hunk 10 above)
Comment (amolnar-rh) [reply modules/ztp-image-based-upgrade-upgrade.adoc:137]: [It is](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/policygenerator-kustomize-plugin/testPolicyGenTemplate/common-ranGen.yaml) and it's a file that is provided in the `ztp-site-generate` container by default so I'd think that the context is clear. However, we don't want to include the full content of the file, only the relevant bits, so I'm putting `[...]` at the beginning and end

**Pull request merged**